cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -4965,8 +4965,16 @@ macro(build_awssdk)
   set(AWSSDK_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/awssdk_ep-install")
   set(AWSSDK_INCLUDE_DIR "${AWSSDK_PREFIX}/include")
 
+  # The AWS SDK has a few warnings around shortening lengths
+  set(AWS_EP_COMMON_CMAKE_ARGS ${EP_COMMON_CMAKE_ARGS})
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_ID STREQUAL
+                                                    "Clang")
+    string(REPLACE "-Wconversion" "-Wconversion -Wno-error=shorten-64-to-32"
+                   AWS_EP_COMMON_CMAKE_ARGS "${AWS_EP_COMMON_CMAKE_ARGS}")
+  endif()
+
   set(AWSSDK_COMMON_CMAKE_ARGS
-      ${EP_COMMON_CMAKE_ARGS}
+      ${AWS_EP_COMMON_CMAKE_ARGS}
```

Comment:
Could you use "appending `-Wno-error*`" approach instead of "replacing a flag" approach like we did for gRPC?

https://github.com/apache/arrow/blob/49be60f5c424cca40bbc5a6d1948ad7e800afaab/cpp/cmake_modules/ThirdpartyToolchain.cmake#L3920-L3940

Reply:
Ah, yes! Thanks for this pointer. I tried constructing this before based on the substrait approach which is similar, but I couldn't get the cmake to work, but this did!